### PR TITLE
Proposal to support references to schemas in other files

### PR DESCRIPTION
**Pull request by sw23**
I had to update the package to resolve references across files to work with some of our schemas, so wanted to feed the solution back to the project.  I also added a test case highlighting some of the corner cases possible with references.

### CODE REVIEW

**Comment by sw23** [comment]
@dblanchette - Mentioning in case you hadn't seeing this.  Can you review when you have some time?  Thanks

**Comment by dblanchette** [comment]
Sorry, I did not saw it indeed. Will look at this today.

File: json_schema_for_humans/generate.py [hunk 1]
```diff
@@ -328,7 +345,8 @@ def generate_from_file_object(
     """Generate the JSON schema documentation from opened file objects for both input and output files. The
     result_file should be opened in write mode.
     """
-    result = generate_from_schema(json.load(schema_file), minify, deprecated_from_description, default_from_description)
+    schema_path = os.path.abspath(schema_file.name).split(os.path.sep)
```
Comment (dblanchette) [inline json_schema_for_humans/generate.py]: Unfortunately that does not work on Windows because `os.path.sep` is `\`, but the references in the schemas are always with forward slashes. I tested the following:
```suggestion
    schema_path = list(Path(schema_file.name).resolve().parts)
```
Don't forget to add the import `from pathlib import Path` at line 5. I cannot make that a suggestion, because it is too far from any code modified in the current PR.

File: tests/generate_test.py [hunk 2]
```diff
@@ -9,21 +9,23 @@
 
 def _get_test_case(name: str) -> Dict[str, Any]:
```
Comment (dblanchette) [inline tests/generate_test.py]:
```suggestion
def _get_test_case(name: str) -> Tuple[Dict[str, Any], List[str]]:
```
Along with adding `Tuple` at line 3
```
from typing import Any, Dict, List, Tuple
```

File: json_schema_for_humans/generate.py (hunk 1 above)
Comment (sw23) [reply json_schema_for_humans/generate.py]: Agreed that all references in schemas should be forward slashes.  Inside of `generate_from_file_object` and `generate_from_filename` we should be dealing with the schema file passed via the command line (or test), which should have an OS-specific path (and `os.path.abspath(...)` will always return a path delimited by `os.path.sep`).  I ran it on Windows (cmd and powershell (and Linux for a sanity check)) and confirmed it worked as expected.  That said, I'm fine switching to pathlib if you'd like.  The only gotcha I see is that we need schema_path to be an absolute path; otherwise if schemas have refs with more '../'s than the relative path specified, we won't be able to resolve the reference.

File: tests/generate_test.py (hunk 2 above)
Comment (sw23) [reply tests/generate_test.py]: Ah, thanks - I don't use type hints much so did my best to go with the flow :-)  I'll fix

File: json_schema_for_humans/generate.py (hunk 1 above)
Comment (dblanchette) [reply json_schema_for_humans/generate.py]: You are right that it works, there was something wrong with my test.

I'm not sure I understand the last part of your message correctly, but the `.resolve()` part makes the path absolute.

It's probably best to keep it the way you did at first as it is simpler.

File: tests/generate_test.py (hunk 2 above)
Comment (dblanchette) [reply tests/generate_test.py]: Type hints + mypy saved us from a lot of issues on a bigger project (but they can sometimes be a pain :P)

Comment (sw23) [reply tests/generate_test.py]: Should be all set now

File: json_schema_for_humans/generate.py (hunk 1 above)
Comment (sw23) [reply json_schema_for_humans/generate.py]: Ah ok, thanks. If `.resolve()` produces an absolute path then I think either solution would work.

**Comment by dblanchette** [comment]
Released in version 0.4. Thanks again!
